### ISSUE TITLE
Set message property "Properties"

### ISSUE THREAD

**Issue by VolkerKamin** [Azure/iothub-explorer#12]
A message object has a property "Properties" that maps from string key to string value.
Is it possible to send a message with user defined Properties["SomeKey"]="SomeValue" to a device using iothub-explorer?

**Comment by pierreca** [comment]
Technically it should be possible: if your JSON contains a top-level `messageId` property then it is considered a `Message` object and sent as is, otherwise it's just considered to be the payload and it's built into a message that doesn't have properties.
This being said, the properties structure is a little unnatural (it's a legacy thing and we don't want to introduce breaking changes) so your JSON might look very weird especially from the command line.
Something along the lines of:
```json
{
  "messageId": "some-uuid",
  "body": "some payload",
  "properties": {
    "propertyList": [
      {"key": "key1", "value": "value1"},
      {"key": "key2", "value": "value2"}
    ]
  }
}
```

This being said, I'd be interested in hearing if you'd like to see a better way to do this with iothub-explorer and why you'd think it's more interesting to have this in iothub-explorer than in some sample code in the SDK repository (https://github.com/azure/azure-iot-sdk-node)

**Comment by pierreca** [comment]
I just checked in a change that fixes the function we use to check if what is passed on the command line is a payload or a message.

There is a way to send messages with properties, by formatting an object on the command line that has a `messageId` property. When this property is set, the code assume the object describes a full message and therefore you can add properties. the syntax is a little weird though, because of the properties structure, and of course you'll have to escape that properly depending on your shell. Here's how a message should look like:

```json
{
  "messageId": "some id",
  "data": "message body",
  "properties": {
    "propertyList": [
      { "key": "some_key", "value": "some_value" },
      { "key": "some_other_key", "value": "some_other_value" }
    ]
  }
}
```

We will release a new package that contain this bugfix.

**Comment by pierreca** [comment]
Fix has been released with [iothub-explorer@1.1.9](https://www.npmjs.com/package/iothub-explorer). Thanks!
